sharepoint-server-farm-ha/azuredeploy.json

```diff
@@ -0,0 +1,1985 @@
+						"Configuration":"[concat('{\"roles\":[{\"type\":\"application\",\"properties\":{\"fqdn\":\"',reference(variables('spCAIPAddressName')).dnsSettings.fqdn,'\",\"port\":',variables('spCentralAdminPort'),'}}],\"configureForHa\":\"True\",\"loadBalancedSetProbePort\":',variables('spWebProbePort'),'}')]"
+					}
+				},
+				"protectedSettings": {
+					"Items": {
+						"AdminPassword": "[parameters('adminPassword')]",
+						"SharePointSetupUserAccountPassword": "[parameters('sharePointSetupUserAccountPassword')]",
+						"SharePointFarmAccountPassword": "[parameters('sharePointFarmAccountPassword')]",
+						"SharePointFarmPassphrasePassword": "[parameters('sharePointFarmPassphrasePassword')]",
+						"SqlServerServiceAccountPassword": "[parameters('sqlServerServiceAccountPassword')]"
+					}
+				}
+			}
+		},
+		{
+			"type": "Microsoft.Compute/virtualMachines/extensions",
+			"name": "[concat(variables('spAppVMName'),'1/ConfigureSPAppServer')]",
+			"apiVersion": "2015-05-01-preview",
+			"location": "[parameters('location')]",
+			"dependsOn": [
+				"[concat('Microsoft.Compute/virtualMachines/', variables('spAppVMName'),'1')]",
+				"[concat('Microsoft.Compute/virtualMachines/', variables('spAppVMName'),'0/extensions/ConfigureSPAppServer')]"
+			],
+			"properties": {
+				"publisher": "Microsoft.Powershell",
+				"type": "DSC",
+				"typeHandlerVersion": "1.7",
+				"settings": {
+					"ModulesUrl": "[variables('spModulesURL')]",
+					"ConfigurationFunction": "[variables('spConfigurationFunction')]",
+					"Properties": {
+						"DomainName": "[parameters('domainName')]",
+						"AdminCreds":{
+							"UserName": "[parameters('adminUserName')]",
+							"Password": "PrivateSettingsRef:AdminPassword"
+						},
+						"SharePointSetupUserAccountcreds":{
+							"UserName": "[parameters('sharePointSetupUserAccountUserName')]",
+							"Password": "PrivateSettingsRef:SharePointSetupUserAccountPassword"
+						},
+						"SharePointFarmAccountcreds":{
+							"UserName": "[parameters('sharePointFarmAccountUserName')]",
+							"Password": "PrivateSettingsRef:SharePointFarmAccountPassword"
+						},
+						"SharePointFarmPassphrasecreds":{
+							"UserName": "ignore",
+							"Password": "PrivateSettingsRef:SharePointFarmPassphrasePassword"
+						},
+						"SQLServiceCreds":{
+							"UserName": "[parameters('sqlServerServiceAccountUserName')]",
+							"Password": "PrivateSettingsRef:SqlServerServiceAccountPassword"
+						},
+						"DatabaseName":"[parameters('configDatabaseName')]",
+						"AdministrationContentDatabaseName":"[parameters('administrationContentDatabaseName')]",
+						"DatabaseServer":"[concat(variables('sqlVMName'),'1.',parameters('domainName'))]",
+						"SqlAlwaysOnAvailabilityGroupName":"[variables('sqlAOAGName')]",
+						"PrimaryReplica":"[concat(variables('sqlVMName'),'1')]",
+						"SecondaryReplica":"[concat(variables('sqlVMName'),'0')]",
+						"databaseNames":null,
```

Comment:
I take it the secondary has to have this null, but not the primary. Can you have one template and call it with two different databaseNames properties? You could even have an array in the variables section and base whether this is null on the copyIndex.